src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -73,6 +75,9 @@ public partial class MarkStep : IStep
 		// Stores, for compiler-generated methods only, whether they require the reflection
 		// method body scanner.
 		readonly Dictionary<MethodBody, bool> _compilerGeneratedMethodRequiresScanner;
+		private readonly Dictionary<TypeDefinition, TypeDefinitionDependencyNode> _typeNodes = new ();
+		private readonly Dictionary<MethodDefinition, MethodDefinitionDependencyNode> _methodNodes = new ();
+		private readonly Dictionary<TypeDefinition, TypeIsRelevantToVariantCastingNode> _typeIsRelevantToVariantCastingNodes = new ();
```

Comment:
Good point, moved those to the NodeFactory. I just copied the NodeCache definition since it's in ILCompiler.Compiler.csproj and its containing type has ilc-specific code.